src/middleware.js

```diff
@@ -37,28 +29,30 @@ function init(swaggerPath, options) {
-
+                    const isOpenApi3 = dereferenced.openapi === '3.0.0';
                     const parameters = dereferenced.paths[currentPath][currentMethod].parameters || [];
-
-                    let bodySchema = middlewareOptions.expectFormFieldsInBody
-                        ? parameters.filter(function (parameter) { return (parameter.in === 'body' || (parameter.in === 'formData' && parameter.type !== 'file')) })
-                        : parameters.filter(function (parameter) { return parameter.in === 'body' });
-
-                    if (makeOptionalAttributesNullable) {
-                        schemaPreprocessor.makeOptionalAttributesNullable(bodySchema);
-                    }
-                    if (bodySchema.length > 0) {
-                        const validatedBodySchema = _getValidatedBodySchema(bodySchema);
-                        schemas[parsedPath][currentMethod].body = buildBodyValidation(validatedBodySchema, dereferenced.definitions, swaggers[1], currentPath, currentMethod, parsedPath);
+                    if (isOpenApi3){
+                        schemas[parsedPath][currentMethod].body = swagger3.buildV3BodyValidation(dereferenced, swaggers[1], currentPath, currentMethod, middlewareOptions);
+                    } else {
+                        let bodySchema = middlewareOptions.expectFormFieldsInBody
+                            ? parameters.filter(function (parameter) { return (parameter.in === 'body' || (parameter.in === 'formData' && parameter.type !== 'file')) })
+                            : parameters.filter(function (parameter) { return parameter.in === 'body' });
+                        if (makeOptionalAttributesNullable) {
+                            schemaPreprocessor.makeOptionalAttributesNullable(bodySchema);
+                        }
+                        if (bodySchema.length > 0) {
+                            const validatedBodySchema = swagger2.getValidatedBodySchema(bodySchema);
+                            schemas[parsedPath][currentMethod].body = swagger2.buildBodyValidation(validatedBodySchema, dereferenced.definitions, swaggers[1], currentPath, currentMethod, parsedPath, middlewareOptions);
+                        }
                     }
 
                     let localParameters = parameters.filter(function (parameter) {
                         return parameter.in !== 'body';
                     }).concat(pathParameters);
 
                     if (localParameters.length > 0 || middlewareOptions.contentTypeValidation) {
-                        schemas[parsedPath][currentMethod].parameters = buildParametersValidation(localParameters,
-                            dereferenced.paths[currentPath][currentMethod].consumes || dereferenced.paths[currentPath].consumes || dereferenced.consumes);
+                        schemas[parsedPath][currentMethod].parameters = swagger2.buildParametersValidation(localParameters,
```

Comment:
if it's not specific to swagger2 and also being used by both of them don't put it under swagger2